bin/cliain/src/contracts.rs

```diff
@@ -0,0 +1,306 @@
+use crate::{commands::ContractOptions, Command, ContractMessageTranscoder};
+use aleph_client::{send_xt, wait_for_event, AnyConnection, SignedConnection};
+use anyhow::anyhow;
+use codec::{Compact, Decode};
+use contract_metadata::ContractMetadata;
+use log::{debug, info};
+use serde::{Deserialize, Serialize};
+use sp_core::{Pair, H256};
+use std::{
+    fs::{self, File},
+    path::Path,
+};
+use substrate_api_client::{compose_extrinsic, AccountId, GenericAddress, XtStatus};
+
+#[derive(Debug, Decode, Clone)]
+pub struct ContractCodeRemovedEvent {
+    code_hash: H256,
+}
+
+#[derive(Debug, Decode, Clone)]
+pub struct ContractInstantiatedEvent {
+    deployer: AccountId,
+    contract: AccountId,
+}
+
+#[derive(Debug, Decode, Clone)]
+pub struct ContractCodeStoredEvent {
+    code_hash: H256,
+}
+
+#[derive(Debug, Decode, Clone, Serialize, Deserialize)]
+pub struct InstantiateWithCodeReturnValue {
+    pub contract: AccountId,
+    pub code_hash: H256,
+}
+
+fn storage_deposit(storage_deposit_limit: Option<u128>) -> Option<Compact<u128>> {
+    storage_deposit_limit.map(Compact)
+}
+
+pub fn upload_code(
+    signed_connection: SignedConnection,
+    command: Command,
```

Comment:
In this function and others instead of accepting `Enum` in the signature accept required args (or use some wrapper struct for them).

Reply:
done